test/unit/remote-config/remote-config-api-client.spec.ts

```diff
@@ -159,4 +190,120 @@ describe('RemoteConfigApiClient', () => {
+            }
+          });
+        });
+    });
+
+    it('should reject when the etag is not present', () => {
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .resolves(utils.responseFrom(testResponse));
+      stubs.push(stub);
+      const expected = new FirebaseRemoteConfigError('invalid-argument', 'ETag header is not present in the server response.');
+      return apiClient.validateTemplate(remoteConfigTemplate)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject when a full platform error response is received', () => {
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom(ERROR_RESPONSE, 404));
+      stubs.push(stub);
+      const expected = new FirebaseRemoteConfigError('not-found', 'Requested entity not found');
+      return apiClient.validateTemplate(remoteConfigTemplate)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject unknown-error when error code is not present', () => {
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom({}, 404));
+      stubs.push(stub);
+      const expected = new FirebaseRemoteConfigError('unknown-error', 'Unknown server error: {}');
+      return apiClient.validateTemplate(remoteConfigTemplate)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject unknown-error for non-json response', () => {
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom('not json', 404));
+      stubs.push(stub);
+      const expected = new FirebaseRemoteConfigError(
+        'unknown-error', 'Unexpected response with status: 404 and body: not json');
+      return apiClient.validateTemplate(remoteConfigTemplate)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject with failed-precondition when a validation error occurres', () => {
+      const invalidTemplateError = {
+        error: {
+          code: 400,
+          message: "[VALIDATION_ERROR]: [foo] are not valid condition names. All keys in all conditional value maps must be valid condition names.",
+          status: "FAILED_PRECONDITION"
+        }
+      };
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom(invalidTemplateError, 400));
+      stubs.push(stub);
+      const expected = new FirebaseRemoteConfigError('failed-precondition',
+        '[VALIDATION_ERROR]: [foo] are not valid condition names. All keys in all conditional value maps must be valid condition names.');
+      return apiClient.validateTemplate(remoteConfigTemplate)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject with failed-precondition when a validation error occurres', () => {
```

Comment:
Combine with the previous test case, via an array.

```
const errorMessages = [...];
errorMessages.forEach((message) => {
  it(....)
});
```

Reply:
Thanks! Done.